### PR TITLE
Linting, building and automated publishing

### PR DESCRIPTION
**Pull request by knopki**
`grunt`, custom nodejs http server, `pm2` configuration removed.

`.eslintrc` from upstream with some added globals. Some files edited to make `eslint` happy. @Igor89 please test this files:
- `Sim900r.js` (remove: unused callbacks, escape character)
- `dweetio.js` (fix bug with always empty response)
- `octoliner.js` (make `i` shared)
- `quaddisplay.js ` (see `interval`)
- `usb-keyboard.js` (remove unused callbacks, make `i` shared)
- `wifi.js` (`self` magic, useless function argument)
- `x-fet.js` (make `i` shared)

`prettier` initialized with empty config and all modules prettyfied. May be we need set some options for `prettier`? (`.prettierrc` needed even if empty to force `vscode` use empty config instead of defaults of extension)

CircleCI automation:
- lint (eslint and prettylint)
- publish to GCS bucket when merged to master

@Igor89 what is http://js.amperka.ru/emu/? If it used we can just put it to the GCS (already did) or publish source code as part of this repository.

@nkrkv @Igor89 BTW Espruino supports [URLs in `require`()](http://www.espruino.com/Modules#from-github-or-anywhere-on-the-internet-) (but I think example is broken because of wrong urls). So any module can be hosted from `github` without special repositories. Just to know.

### CODE REVIEW

**Comment by knopki** [comment]
Small update: only `modules` dir prettified. `json` excluded because of autogenerated nature of `ISKRAJS.json`.

**Comment by knopki** [comment]
@yurybotov if и for должны отделяться пробелом однозначно. Не вижу, где без пробела?

**Comment by knopki** [comment]
image

**Comment by nkrkv** [comment]
@Igor89 review?

File: modules/@amperka/meteo-sensor.js [hunk 1]
```diff
@@ -69,29 +71,269 @@ function TroykaMeteoSensor(opts) {
...
-      248, 201, 154, 171, 60, 13, 94, 111, 65, 112, 35, 18, 133, 180, 231, 214,
-      122, 75, 24, 41, 190, 143, 220, 237, 195, 242, 161, 144, 7, 54, 101, 84,
-      57, 8, 91, 106, 253, 204, 159, 174, 128, 177, 226, 211, 68, 117, 38, 23,
-      252, 205, 158, 175, 56, 9, 90, 107, 69, 116, 39, 22, 129, 176, 227, 210,
-      191, 142, 221, 236, 123, 74, 25, 40, 6, 55, 100, 85, 194, 243, 160, 145,
-      71, 118, 37, 20, 131, 178, 225, 208, 254, 207, 156, 173, 58, 11, 88, 105,
-      4, 53, 102, 87, 192, 241, 162, 147, 189, 140, 223, 238, 121, 72, 27, 42,
-      193, 240, 163, 146, 5, 52, 103, 86, 120, 73, 26, 43, 188, 141, 222, 239,
-      130, 179, 224, 209, 70, 119, 36, 21, 59, 10, 89, 104, 255, 206, 157, 172
+      0,
```
Comment (nkrkv) [inline modules/@amperka/meteo-sensor.js]: IMO, this array literal might be protected with an ESLint comment directive to stay compact

File: modules/@amperka/usb-keyboard.js [hunk 2]
```diff
@@ -3,37 +3,68 @@
...
-    0x15, 0x00, // Logical Minimum (0),
-    0x25, 0x68, // Logical Maximum(104),
-    0x05, 0x07, // Usage Page (Key Codes),
-    0x19, 0x00, // Usage Minimum (0),
-    0x29, 0x68, // Usage Maximum (104),
-    0x81, 0x00, // Input (Data, Array),
+    0x05,
+    0x01, // Usage Page (Generic Desktop),
+    0x09,
+    0x06, // Usage (Keyboard),
```
Comment (nkrkv) [inline modules/@amperka/usb-keyboard.js]: USB descriptors are byte pairs (words), so I’d leave this array literal as is (ESLint exception)

**Comment by nkrkv** [comment]
> http://js.amperka.ru/emu/ Do we use it?

AFAIK no

File: modules/@amperka/meteo-sensor.js (hunk 1 above)
Comment (knopki) [reply modules/@amperka/meteo-sensor.js]: It's `prettier` thing, fixed.

File: modules/@amperka/usb-keyboard.js (hunk 2 above)
Comment (knopki) [reply modules/@amperka/usb-keyboard.js]: It's `prettier` thing, fixed.

**Comment by knopki** [comment]
@nkrkv `singleQuote` rule added with array literals ignore. Changeset becomes from `+4,201 −1,457` to `+3,508 −1,033`.

**Comment by knopki** [comment]
@Igor89 Am I understand properly that `wifi.js` and `dweetio.js` works but problem only examples (outside of this repo)?

Also, about `keyboard.js`. This version doesn't work, but previous does? I mean my internal brain JS-compiler says that changes are trivial and can't broke anything. So is it me or previous changes or regression with new version of espruino?

**Comment by Igor89** [comment]
@knopki That's right.

> Am I understand properly that wifi.js and dweetio.js works but problem only examples (outside of this repo)?

 That's right.

Module `keyboard.js ` on the previous version also does not work. But this is not related to your changes. `keyboard.js ` must be corrected in another PR.

**Comment by knopki** [comment]
@nkrkv @Igor89 Just to note: I don't have rights to merge this PR :)

**Pull request merged**